pkg/etcd/etcd.go

```diff
@@ -1162,46 +1194,67 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
-func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client corev1.CoreV1Interface, memberName string, promoted bool) (*v1.Node, []byte, error) {
-	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
+func (e *ETCD) getETCDStatus(ctx context.Context, url string) (*clientv3.StatusResponse, error) {
+	ctx, cancel := context.WithTimeout(ctx, defaultDialTimeout)
+	defer cancel()
+
+	return e.client.Status(ctx, url)
+}
```

Comment:
You're not doing anything with this return. I would probably just return an error if it not OK, and nil otherwise. As it currently stands this is a low-value function, as all it does is wrap a Status call with a context timeout.

```suggestion
func (e *ETCD) checkETCDStatus(ctx context.Context, url string) error {
	ctx, cancel := context.WithTimeout(ctx, defaultDialTimeout)
	defer cancel()

	resp ,err := e.client.Status(ctx, url)
	if err != nil {
		return errors.Wrap(err, "failed to check etcd member status")
	}
	if len(resp.Errors) != 0 {
		return errors.New("etcd member has status errors: " + strings.Join(resp.Errors, ","))
	}
	return nil
}
```